projector-plugin/src/main/kotlin/ui/ProjectorStatusWidget.kt

```diff
@@ -0,0 +1,130 @@
+import isProjectorAutoStarting
+import isProjectorDisabled
+import isProjectorRunning
+import java.awt.Component
+import java.awt.event.MouseEvent
+import javax.swing.Icon
+
+class ProjectorStatusWidget(project: Project)
+  : EditorBasedWidget(project),
+    StatusBarWidget.MultipleTextValuesPresentation,
+    StatusBarWidget.Multiframe,
+    ProjectorStateListener {
+
+  override fun ID(): String = ProjectorStatusWidget::class.java.name
+
+  override fun copy(): StatusBarWidget = ProjectorStatusWidget(project)
+
+  override fun getPopupStep(): ListPopup {
+    val context = DataManager.getInstance().getDataContext(myStatusBar as Component)
+    val actionGroup = getProjectorActionGroup(PROJECTOR_TOOLBAR_ACTION_GROUP)
+
+    return JBPopupFactory.getInstance().createActionGroupPopup("Projector",
+                                                               actionGroup,
+                                                               context,
+                                                               false,
+                                                               null,
+                                                               5)
+  }
+
+  override fun getTooltipText(): String = updateTooltip()
+
+  override fun getClickConsumer(): Consumer<MouseEvent>? = null
+
+  override fun getSelectedValue(): String = updateText()
+
+  override fun getPresentation(): WidgetPresentation = this
+
+  override fun getIcon() = updateIcon()
+
+  override fun install(statusBar: StatusBar) {
+    super.install(statusBar)
+    ProjectorService.subscribe(this)
+  }
+
+  override fun dispose() {
+    ProjectorService.unsubscribe(this)
+    super.dispose()
+  }
+
+  fun update() = myStatusBar.updateWidget(ID())
+
+  override fun stateChanged() = update()
+
+  private fun updateIcon(): Icon {
+    return when {
+      isActivationNeeded() -> RED_DOT
+      isProjectorRunning() -> GREEN_DOT
+      isProjectorAutoStarting() -> YELLOW_DOT
+      isProjectorDisabled() -> BLACK_DOT
+      else -> BLACK_DOT
+    }
+  }
+
+  private fun updateText() = "Projector"
+
+  private fun updateTooltip(): String {
+    return when {
+      isActivationNeeded() -> "Activation is needed"
+      isProjectorRunning() -> "Projector is running"
+      isProjectorAutoStarting() -> "Projector is starting"
+      isProjectorDisabled() -> "Projector is disabled"
+      else -> "Impossible state"
+    }
+  }
+
+  companion object {
+    private const val PROJECTOR_TOOLBAR_ACTION_GROUP = "projector.toolbar"
+
+    @JvmField
+    val RED_DOT = IconLoader.getIcon("/META-INF/redSign.svg", ProjectorStatusWidget::class.java)
+
+    @JvmField
+    val GREEN_DOT = IconLoader.getIcon("/META-INF/greenSign.svg", ProjectorStatusWidget::class.java)
+
+    @JvmField
+    val YELLOW_DOT = IconLoader.getIcon("/META-INF/yellowSign.svg", ProjectorStatusWidget::class.java)
+
+    @JvmField
+    val BLACK_DOT = IconLoader.getIcon("/META-INF/blackSign.svg", ProjectorStatusWidget::class.java)
```

Comment:
Two concerns:
1. Is it right to load them statically? I can think of lazy loading, but it seems it's not right too because it can be loaded in AWT thread and UI will freeze. So maybe your variant is OK, I'm just asking.
2. Can we maybe avoid assets and draw these dots programmatically? Like creating a BufferedImage and drawing a circle there?

Reply:
Tanks for review. All status icons lazily initialized and drawn in code now.